packages/core/src/external/cda/partition-payload.ts

```diff
@@ -0,0 +1,81 @@
+import { toArray } from "@metriport/shared";
+import { createXMLParser } from "@metriport/shared/common/xml-parser";
+import { sizeInBytes } from "../../util/string";
+import { XMLBuilder } from "fast-xml-parser";
+
+const MAX_CHUNK_SIZE = 10_000_000; // 10MB in bytes
+
+export function partitionPayload(payloadRaw: string): string[] {
+  if (sizeInBytes(payloadRaw) < MAX_CHUNK_SIZE) return [payloadRaw];
+
+  const parser = createXMLParser({
+    ignoreAttributes: false,
+    attributeNamePrefix: "_",
+    removeNSPrefix: true,
+  });
+  const json = parser.parse(payloadRaw);
+
+  if (!json.ClinicalDocument?.component?.structuredBody?.component) {
+    return [payloadRaw];
+  }
+
+  const components = toArray(json.ClinicalDocument?.component?.structuredBody?.component);
+
+  const chunks: string[] = [];
+  let currentComponents: string[] = [];
+  let currentSize = 0;
+
+  const builder = new XMLBuilder({
+    format: false,
+    ignoreAttributes: false,
+    attributeNamePrefix: "_",
+    suppressEmptyNode: true,
+    suppressBooleanAttributes: false,
+  });
+
+  function createChunk(comps: string[]) {
+    const chunk = {
+      ...json,
+      ClinicalDocument: {
+        ...json.ClinicalDocument,
+        component: {
+          ...json.ClinicalDocument.component,
+          structuredBody: {
+            ...json.ClinicalDocument.component.structuredBody,
+            component: comps,
+          },
+        },
+      },
+    };
+    return builder.build(chunk);
+  }
+
+  for (const component of components) {
```

Comment:
Can we rename the consts/variables to make it eaiser to read the code, please?
- component -> currentComponent
- componentSize -> currentSize
- currentComponents -> groupedComponents
- currentSize -> groupedSize

Reply:
yea, this is much better!